core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala

```diff
@@ -137,6 +139,17 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
     )
   }
 
+  /** If Tasks is too large, remove and garbage collect old tasks */
+  private def trimTasksIfNecessary(taskData: HashMap[Long, TaskUIData]) = synchronized {
+    if (taskData.size > retainedTasks) {
+      val toRemove = (taskData.size - retainedTasks)
+      val oldIds = taskData.map(_._2.taskInfo.taskId).toList.sorted.take(toRemove)
```

Comment:
is there a reason we are using _._2.taskInfo.taskId instead of just the Long ( _._1), which is the taskId also?

I'm a bit concerned with the sorting here.  If someone is using really large number of tasks , say 1 million I'm afraid this sorting might be expensive, especially if we are doing it for every single task that comes in after hitting the limit, we have to keep resorting every time.